### DISCUSSION TITLE
Formatting java codeblocks in markdown with conform.nvim

### DISCUSSION THREAD

**Discussion by codingluke** [mfussenegger/nvim-jdtls#583] [Q&A]
Hi there, thx a lot for this plugin, I use it on a daliy basis :)

As I am a teacher and have a lot of java code snippets in markdown files, it would be a blessing to be able to format/lint them inline.

I use lazyvim and have [conform.nvim](https://www.lazyvim.org/plugins/formatting#conformnvim) enabled.
conform.nvim is able to format [codeblocks in markdown](https://github.com/stevearc/conform.nvim/blob/master/doc/advanced_topics.md#injected-language-formatting-code-blocks)

The one thing I can not find is, what formatter I have to configure on `formatters_by_ft` so that it takes the one defined by `nvim-jdtls` settings.java.format. On a normal .java file with running Jdtls formatting works fine.

```lua
 formatters_by_ft = {
    java = { "jdtls" } -- does not work
}
```

Thx a lot!

**Comment by mfussenegger** [comment]
I'm not familiar with confirm.nvim but looking at https://github.com/stevearc/conform.nvim it sounds as if you need to enable `lsp_fallback`.

nvim-jdtls/eclipse.jdt.ls is just a language-server that uses the neovim lsp client. It's not a standalone formatter tool

**Comment by codingluke** [comment]
Thank you for the hint. the lsp_fallback is just available for format_on_save. Also the jdtls lsp is not automatically started when editing markdown files. I just use google-java-format now which works as expected. The good part of the jdtls formatting is, that it is configurabe with eclipse/intellij xml schemas. Fortunately for my markdown use-case this is not needed.
